osquery/tables/networking/curl_certificate.cpp

```diff
@@ -335,49 +352,133 @@ Status getTLSCertificate(const std::string& hostname,
     return Status::failure("Failed to create OpenSSL CTX object");
   }
 
-  auto delBIO = [](BIO* bio) { BIO_free_all(bio); };
-  auto server = std::unique_ptr<BIO, decltype(delBIO)>(
-      BIO_new_ssl_connect(ctx.get()), delBIO);
-  if (server == nullptr) {
-    return Status::failure("Failed to create OpenSSL BIO object");
+  std::string port = "443";
+  auto connect_hostname = hostname;
+  auto delim = hostname.find(":");
+  if (delim + 1 == hostname.length()) {
+    // if no port specified use default port
+    connect_hostname = hostname.substr(0, delim);
+  } else if (delim != std::string::npos) {
+    port = hostname.substr(delim + 1, std::string::npos);
+    connect_hostname = hostname.substr(0, delim);
+  }
+
+  int ret;
+#ifdef WIN32
+  // Initialize Winsock
+  WSADATA wsaData;
+  ret = WSAStartup(MAKEWORD(2, 2), &wsaData);
+  if (ret != 0) {
+    return Status::failure("WSAStartup failed with error: " +
+                           std::to_string(ret));
   }
+#endif
 
-  std::string port = ":443";
-  auto ext_hostname = hostname;
-  auto conn_hostname = hostname;
-  auto delim = hostname.find(":");
-  if (delim == std::string::npos) {
-    conn_hostname += ":443";
-  } else {
-    ext_hostname = hostname.substr(0, delim);
+  // connect up the socket first and then pass to BIO
+  struct addrinfo hints, *addrinfo, *current_addr, *preferred_addr;
+  memset(&hints, 0, sizeof(hints));
+  hints.ai_family = PF_UNSPEC;
+  hints.ai_socktype = SOCK_STREAM;
+  hints.ai_flags = AI_NUMERICSERV | AI_ADDRCONFIG;
+
+  ret = getaddrinfo(connect_hostname.c_str(), port.c_str(), &hints, &addrinfo);
```

Comment:
Also consider a scope guard that contains the logic 
```cpp
    if (addrinfo != nullptr) {
      freeaddrinfo(addrinfo);
    }
```
you will have to include `[&addrinfo]` in the capture group.